registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/service/impl/SyncRegistrationServiceImpl.java

```diff
@@ -1,745 +1,821 @@
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see io.mosip.registration.processor.status.service.SyncRegistrationService#
+	 * getByIds(java.util.List)
+	 */
+	@Override
+	public List<RegistrationStatusDto> getByIds(List<RegistrationStatusSubRequestDto> requestIds) {
+		regProcLogger.debug(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.USERID.toString(), "",
+				"SyncRegistrationServiceImpl::getByIds()::entry");
+
+		try {
+			List<String> registrationIds = new ArrayList<>();
+
+			for (RegistrationStatusSubRequestDto registrationStatusSubRequestDto : requestIds) {
+				registrationIds.add(registrationStatusSubRequestDto.getRegistrationId());
+			}
+			if (!registrationIds.isEmpty()) {
+				List<SyncRegistrationEntity> syncRegistrationEntityList = syncRegistrationDao.getByIds(registrationIds);
+
+				regProcLogger.debug(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.USERID.toString(), "",
+						"SyncRegistrationServiceImpl::getByIds()::exit");
+				return convertEntityListToDtoListAndGetExternalStatus(syncRegistrationEntityList);
+			}
+			return null;
+		} catch (DataAccessLayerException e) {
+
+			regProcLogger.error(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.REGISTRATIONID.toString(),
+					"", e.getMessage() + ExceptionUtils.getStackTrace(e));
+			throw new TablenotAccessibleException(
+					PlatformErrorMessages.RPR_RGS_REGISTRATION_TABLE_NOT_ACCESSIBLE.getMessage(), e);
+		}
+
+	}
+
+	/**
+	 * Convert entity list to dto list and get external status.
+	 *
+	 * @param syncRegistrationEntityList
+	 *            the sync registration entity list
+	 * @return the list
+	 */
+	private List<RegistrationStatusDto> convertEntityListToDtoListAndGetExternalStatus(
+			List<SyncRegistrationEntity> syncRegistrationEntityList) {
+		List<RegistrationStatusDto> list = new ArrayList<>();
+		if (syncRegistrationEntityList != null) {
+			for (SyncRegistrationEntity entity : syncRegistrationEntityList) {
+				list.add(convertEntityToDtoAndGetExternalStatus(entity));
+			}
+
+		}
+		return list;
+	}
+
+	/**
+	 * Convert entity to dto and get external status.
+	 *
+	 * @param entity
+	 *            the entity
+	 * @return the registration status dto
+	 */
+	private RegistrationStatusDto convertEntityToDtoAndGetExternalStatus(SyncRegistrationEntity entity) {
+		RegistrationStatusDto registrationStatusDto = new RegistrationStatusDto();
+		registrationStatusDto.setRegistrationId(entity.getRegistrationId());
+		registrationStatusDto.setAdditionalInfoReqId(entity.getAdditionalInfoReqId());
+		registrationStatusDto.setStatusCode(RegistrationExternalStatusCode.UPLOAD_PENDING.toString());
+		return registrationStatusDto;
+	}
+
+	@Override
+	public boolean deleteAdditionalInfo(SyncRegistrationEntity syncEntity) {
+		return syncRegistrationDao.deleteAdditionalInfo(syncEntity);
+	}
+
+	@Override
+	public List<PacketStatusDTO> getByPacketIdsWithStatus(List<PacketStatusSubRequestDTO> packetIdsRequested) {

```

Comment:
We can remove this sequence of method and do this whole logic in the controller / packet status service itself? Since it is just the packets ids not in registration table is give a default status code ?

Reply:
fixed